frontend/src/app/offers/create-offer/create-offer.component.ts

```diff
@@ -0,0 +1,154 @@
+      volunteersLimit: [],
+      reserveRecruitmentStartDate: [],
+      reserveRecruitmentEndDate: [],
+      reserveVolunteersLimit: [],
+      description: ['', Validators.required],
+      timeCommitment: ['', Validators.required],
+      benefits: ['', Validators.required],
+      requirements: ['', Validators.required],
+      image: [],
+    })
+
+      this.userSubscription = this.authService.user$
+      .subscribe(
+        response => {
+          this.user = response;
+            this.isAdmin = this.user['isAdministrator'];
+            this.hasOrganization = this.user['organizations'].length > 0;
+          }
+      );
+
+    this.route.params
+    .map(params => params.offerId)
+    .filter(offerId => offerId !== undefined)
+    .switchMap(offerId => this.offersService.getOffer(offerId))
+    .do(offer => this.toDataUrl(offer.image, (base64) => this.offer.image = {
+      content: base64,
+      filename: 'image.jpg'
+    }))
+    .subscribe((response: BaseOffer) => {
+      this.offer = response as AppOffer;
+      this.inEditMode = true;
+      this.form.patchValue(this.offer);
+    });
+  }
+
+  ngOnDestroy() {
+    if (this.userSubscription) {
+      this.userSubscription.unsubscribe();
+    }
+  }
+
+  toDataUrl(url, callback) {
+    const xhr = new XMLHttpRequest();
+    xhr.onload = function() {
+        const reader = new FileReader();
+        reader.onloadend = function() {
+            callback(reader.result);
+        }
+        reader.readAsDataURL(xhr.response);
+    };
+    xhr.open('GET', url);
+    xhr.responseType = 'blob';
+    xhr.send();
+}
+
+  onSubmit(offer: AppOffer) {
+    // TODO - delete those when we decide what date format we want to have
+    offer.startedAt = offer.startedAt + 'T00:00:00Z';
+    offer.finishedAt = offer.finishedAt + 'T00:00:00Z';
+
+    if (!offer.recruitmentEndDate) {
+      offer.recruitmentEndDate = null
+    } else {
+      offer.recruitmentEndDate = offer.recruitmentEndDate + 'T00:00:00Z';
+    }
+    if (!offer.recruitmentStartDate) {
+      offer.recruitmentStartDate = null
+    } else {
+      offer.recruitmentStartDate = offer.recruitmentStartDate + 'T00:00:00Z';
+    }
+    if (!offer.reserveRecruitmentEndDate) {
+      offer.reserveRecruitmentEndDate = null
+    } else {
+      offer.reserveRecruitmentEndDate = offer.reserveRecruitmentEndDate + 'T00:00:00Z';
+    }
+    if (!offer.reserveRecruitmentStartDate) {
+      offer.reserveRecruitmentStartDate = null
+    } else {
+      offer.reserveRecruitmentStartDate = offer.reserveRecruitmentStartDate + 'T00:00:00Z';
+    }
+
+    this.form.value.image = this.offer.image;
+    if (this.inEditMode) {
+      this.offersService.editOffer(this.form.value, offer.id)
+      .subscribe();
+    } else {
+      this.offersService.createOffer(this.form.value)
+      .subscribe();
+    }
+  }
+
+  onFileSelected(event) {
+    this.file = event.target.files[0];
+    this.reader.readAsDataURL(this.file)
```

Comment:
This line should go after this.reader.onloadend.

Explanation: First we should register the event handler (onloadend), then run a method that invokes the event. If we leave it like that, it is theoretically possible that the file load will finish before the event handler get registered and no image processing will happen.